### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by nedtwigg**
This PR fixes the deprecation warnings noted in

- #152 

It does so by simply

- copy-pasting `org.gradle.util.ConfigureUtil` into the source of this project
- removing the `logDeprecation` calls
- removing all unused methods

This `ConfigureUtil` class adds about ~100 lines to the project. Unfortunately, even after merging this PR, there are still internal packages which Gradle may change in a future release. For this reason, I would recommend following Gradle's recommendation and removing the `Closure`-based API's in favor of only the `Action` API. I will submit a PR on top of this PR which shows that option.

### CODE REVIEW

**Comment by nedtwigg** [comment]
`./gradlew compatTest` is now passing for me locally.

I maintain a few Gradle plugins myself, and I try quite hard to not bump the minimum requirements on any of them. However, that is trumped by "users should always be able to use the latest Gradle without warnings". If I need to bump the minimum requirements in order to make that possible, I do it. IMO this project ought to bump its minimum reqs, but with this PR you can delay that a bit longer if you'd like :)

File: src/main/java/io/github/gradlenexus/publishplugin/ClosureBackedAction.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import groovy.lang.Closure;
+import org.codehaus.groovy.runtime.metaclass.MissingMethodExecutionFailed;
+import org.gradle.api.Action;
+import org.gradle.api.InvalidActionClosureException;
+import org.gradle.util.Configurable;
+import java.util.Objects;
+
+/**
+ * Exact copy of [org.gradle.util.internal.ClosureBackedAction] from Gradle 7.6, except.
+
```
Comment (szpak) [inline src/main/java/io/github/gradlenexus/publishplugin/ClosureBackedAction.java:28]: Extra line, maybe it breaks the javadoc generation on CI...

**Comment by szpak** [comment]
> If I need to bump the minimum requirements in order to make that possible, I do it. IMO this project ought to bump its minimum reqs, but with this PR you can delay that a bit longer if you'd like :)

I still have some neglected plugins using the old `maven` module, but sometimes Gradle doesn't give you a clear way to support (too many) different versions :-/. If supporting Gradle 6 was problematic, we would switch to 7+. However, then I would like to publish a new version with some new features and bugfixes and the next one bumping the required Gradle versions. So, if keeping Gradle 6+ compatibility (for some time) is achievable, I would prefer to have it (especially that Gradle 8 has just been released and migration to it can take some time).

I propose to wait (a while) for any reply in https://github.com/gradle/gradle/issues/23874 before merging this PR.

**Comment by nedtwigg** [comment]
All CI was failing with

```
> Task :javadoc FAILED
1 error
javadoc: error - No public or protected classes found to document.
```

I worked around this by making a single class public for no good reason. It can be removed completely when converting to `Action` and dropping `Closure`.

File: src/main/java/io/github/gradlenexus/publishplugin/ClosureBackedAction.java (hunk 1 above)
Comment (nedtwigg) [reply src/main/java/io/github/gradlenexus/publishplugin/ClosureBackedAction.java]: Just pushed a fix.

**Pull request closed without merging**